shuttleofx_client/views.py

```diff
@@ -469,60 +470,82 @@ def createUserRepo():
-        call(['git', 'init'])
-        call(['git', 'fetch', '--depth=1', '-n', 'https://github.com/tuttleofx/pluginOFX.git'])
-        output = check_output(['git', 'commit-tree', 'FETCH_HEAD^{tree}', '-m', 'Initial Openfx structure']).strip()
-        call(['git', 'reset', '--hard', output])
-
-        jsonForm = repoFormToJSON(request)
+        check_call(['git', 'init'], cwd=pluginTemplatePath)
+        check_call(['git', 'fetch', '--depth=1', '-n', 'https://github.com/tuttleofx/ofxPluginTemplate.git'], cwd=pluginTemplatePath)
+        output = check_output(['git', 'commit-tree', 'FETCH_HEAD^{tree}', '-m', 'Initial Openfx structure'], cwd=pluginTemplatePath).strip()
+        check_call(['git', 'reset', '--hard', output], cwd=pluginTemplatePath)
+
+        # Replace template data with user's data
+        jsonFile = repoFormToJSON(request)
+        check_call(['python','createPlugin.py', jsonFile, './'], cwd=pluginTemplatePath)
+        # Remove createPlugin.py and the JSON used to replace data with user's data
+        os.remove(os.path.join(pluginTemplatePath,'createPlugin.py'))
+        os.remove(jsonFile)
+
+        check_call(['git', 'add', '-A'], cwd=pluginTemplatePath)
+        check_call(['git', 'commit', '--amend', '-m', 'Initial Openfx structure'], cwd=pluginTemplatePath)
+
+        #Create user's repo
+        req = config.github.post('/user/repos', data={
+            "name": request.form['pluginName'],
+            "private": False,
+        }, format='json')
 
         # Push on user's repo
         owner = req.data['owner']['login']
         url = 'https://' + owner + ':' + session['github_token'][0] + '@github.com' +'/'+ owner +'/'+ request.form['pluginName'] + '.git'
-        call(['git', 'remote', 'add', 'origin', url])
-        call(['git', 'remote', '-v'])
-        call(['git', 'push', 'origin', 'master'])
+        check_call(['git', 'remote', 'add', 'origin', url], cwd=pluginTemplatePath)
+        check_call(['git', 'remote', '-v'], cwd=pluginTemplatePath)
+        check_call(['git', 'push', 'origin', 'master'], cwd=pluginTemplatePath)
 
+        # Delete cloned repo
         shutil.rmtree(pluginTemplatePath)
-        # Get back to old dir
-        os.chdir(cwd)
         status = 'success'
-        message = 'Repository created with success, check your Github account !'
-    else :
-        logging.warning('Error while creating a repo : ' + req.raw_data)
+        message = 'Repository created with success, check your Github account!'
+    except CalledProcessError:
         status = 'error'
-        message = req.data['errors'][0]['message']
+        message = 'There was an error creating your repository, please try again.'
+
     return render_template("createPlugin.html", user=user, provider=provider, status=status, message=message)
 
 @config.g_app.route('/create/sources', methods=['POST'])
 def downloadPluginTemplate():
```

Comment:
Maybe better to create a function to create the repository correctly (from the code in createUserRepo) and then just implement the way to retrieve it (github / archive).